test/jasmine/tests/transform_aggregate_test.js

```diff
@@ -0,0 +1,190 @@
+var Plotly = require('@lib/index');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+var customMatchers = require('../assets/custom_matchers');
+
+describe('aggregate', function() {
+    var gd;
+
+    beforeAll(function() { jasmine.addMatchers(customMatchers);});
+
+    beforeEach(function() { gd = createGraphDiv(); });
+
+    afterEach(destroyGraphDiv);
+
+    it('handles all funcs for numeric data', function() {
+        // throw in some non-numbers, they should get discarded except first/last
+        Plotly.newPlot(gd, [{
+            x: [1, 2, 3, 4, 'fail'],
+            y: [1.1, 2.2, 3.3, 'nope', 5.5],
+            marker: {
+                size: ['2001-01-01', 0.2, 0.1, 0.4, 0.5],
+                color: [2, 4, '', 10, 8],
+                opacity: [0.6, 'boo', 0.2, 0.8, 1.0],
+                line: {
+                    color: [2.2, 3.3, 4.4, 5.5, 'the end']
+                }
+            },
+            transforms: [{
+                type: 'aggregate',
+                groups: ['a', 'b', 'a', 'a', 'a'],
+                aggregations: [
+                    // missing array - the entry is ignored
+                    {array: '', func: 'avg'},
+                    {array: 'x', func: 'sum'},
+                    // non-numerics will not count toward numerator or denominator for avg
```

Comment:
expanded description along with adding a few more functions in https://github.com/plotly/plotly.js/pull/1924/commits/b6ccc01878ebf7edf164576932e2153d8c0088f5